src/EFCore.PG.NodaTime/Infrastructure/Internal/NpgsqlNodaTimeOptionsExtension.cs

```diff
@@ -0,0 +1,40 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using Microsoft.EntityFrameworkCore.Infrastructure;
+using Microsoft.EntityFrameworkCore.Storage;
+using Microsoft.Extensions.DependencyInjection;
+using Npgsql.EntityFrameworkCore.PostgreSQL.Storage.Internal;
+
+namespace Npgsql.EntityFrameworkCore.PostgreSQL.Infrastructure.Internal
+{
+    public class NpgsqlNodaTimeOptionsExtension : IDbContextOptionsExtension
+    {
+        public virtual string LogFragment => "using NodaTime ";
+
+        public virtual bool ApplyServices(IServiceCollection services)
+        {
+            services.AddEntityFrameworkNpgsqlNodaTime();
+
+            return false;
+        }
+
+        public virtual long GetServiceProviderHashCode() => 0;
+
+        public virtual void Validate(IDbContextOptions options)
+        {
+            var internalServiceProvider = options.FindExtension<CoreOptionsExtension>()?.InternalServiceProvider;
+            if (internalServiceProvider != null)
```

Comment:
nit: Invert condition to reduce nesting.

Reply:
As above, I'd rather maintain the code as in the original.